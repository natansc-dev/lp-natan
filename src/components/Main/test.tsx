import { render, screen } from '@testing-library/react'

import { Main } from '.'

describe('<Main />', () => {
  it('Test 001 - should render the heading', () => {
    const { container } = render(
      <Main
        title={'SOU'}
        subtitle={'Fala galera!'}
        description={'TypeScript, ReactJS, NextJS, e Styled Components'}
        typing={['DESENVOLVEDOR', 'GAMER']}
      />
    )

    expect(screen.getByRole('heading', { name: 'SOU' })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Test 002 - should render the span typing', () => {
    const { container } = render(
      <Main
        title={'SOU'}
        subtitle={'Fala galera!'}
        description={'TypeScript, ReactJS, NextJS, e Styled Components'}
        typing={['DESENVOLVEDOR', 'GAMER']}
      />
    )

    expect(screen.queryByTestId('test-span'))

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Test 003 - should render the colors correctly', () => {
    const { container } = render(
      <Main
        title={'SOU'}
        subtitle={'Fala galera!'}
        description={'TypeScript, ReactJS, NextJS, e Styled Components'}
        typing={['DESENVOLVEDOR', 'GAMER']}
      />
    )

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
