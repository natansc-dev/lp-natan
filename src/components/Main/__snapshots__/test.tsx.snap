// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Test 001 - should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: white;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3rem;
}

.c2 {
  width: 100%;
  max-width: 550px;
}

.c2 a:last-child {
  margin-left: 2rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 4.2rem;
  line-height: 4.2rem;
  font-weight: 500;
  margin: 18px 8px 8px 0px;
  color: #fff;
  white-space: nowrap;
  position: relative;
}

.c6 {
  color: var(--main-color);
  font-size: 4.2rem;
  line-height: 4.2rem;
  font-weight: 500;
  margin: 18px 0 8px;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 400;
}

.c7 {
  font-size: 1.6rem;
  margin: 12px 0 22px;
  line-height: 32px;
  font-weight: 400;
}

.c1 {
  width: min(30rem,320px);
  border-radius: 50%;
  border: 10px solid #ffffff1a;
  margin-right: 4rem;
}

.c8 {
  background-color: #fa5b0f;
  color: #FFF;
  border: none;
  border-radius: 26px;
  height: 46px;
  line-height: 46px;
  font-size: 1.6rem;
  padding: 0 32px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c8 span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 span svg {
  font-size: 1.2rem;
  margin-right: 8px;
}

.c9 {
  background-color: transparent;
  color: #fa5b0f;
  border: 1px solid #fa5b0f;
  border-radius: 26px;
  height: 46px;
  line-height: 46px;
  font-size: 1.6rem;
  padding: 0 32px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c9 span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 span svg {
  font-size: 1.2rem;
  margin-right: 8px;
}

<main
  class="c0"
>
  <img
    alt="Um desenvolvedor de frente para um tela com código."
    class="c1"
    src="/images/perfil_b.png"
  />
  <div
    class="c2"
  >
    <h3
      class="c3"
    >
      Fala galera!
    </h3>
    <div
      class="c4"
    >
      <h1
        class="c5"
      >
        SOU
      </h1>
      <span
        class="c6"
        id="test-span"
      >
        <div
          class="Typewriter"
          data-testid="typewriter-wrapper"
        >
          <span
            class="Typewriter__wrapper"
          />
          <span
            class="Typewriter__cursor"
          >
            |
          </span>
        </div>
      </span>
    </div>
    <p
      class="c7"
    >
      <span>
        TypeScript, ReactJS, NextJS, e Styled Components
      </span>
    </p>
    <a
      class="c8"
      color="#FFF"
    >
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"
          />
        </svg>
        MAIS SOBRE MIM
      </span>
    </a>
    <a
      class="c9"
      color="#fa5b0f"
    >
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M320 336c0 8.84-7.16 16-16 16h-96c-8.84 0-16-7.16-16-16v-48H0v144c0 25.6 22.4 48 48 48h416c25.6 0 48-22.4 48-48V288H320v48zm144-208h-80V80c0-25.6-22.4-48-48-48H176c-25.6 0-48 22.4-48 48v48H48c-25.6 0-48 22.4-48 48v80h512v-80c0-25.6-22.4-48-48-48zm-144 0H192V96h128v32z"
          />
        </svg>
        PORTFÓLIO
      </span>
    </a>
  </div>
</main>
`

exports[`<Main /> Test 002 - should render the span typing 1`] = `
.c0 {
  background-color: #06092b;
  color: white;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 3rem;
}

.c2 {
  width: 100%;
  max-width: 550px;
}

.c2 a:last-child {
  margin-left: 2rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 4.2rem;
  line-height: 4.2rem;
  font-weight: 500;
  margin: 18px 8px 8px 0px;
  color: #fff;
  white-space: nowrap;
  position: relative;
}

.c6 {
  color: var(--main-color);
  font-size: 4.2rem;
  line-height: 4.2rem;
  font-weight: 500;
  margin: 18px 0 8px;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 400;
}

.c7 {
  font-size: 1.6rem;
  margin: 12px 0 22px;
  line-height: 32px;
  font-weight: 400;
}

.c1 {
  width: min(30rem,320px);
  border-radius: 50%;
  border: 10px solid #ffffff1a;
  margin-right: 4rem;
}

.c8 {
  background-color: #fa5b0f;
  color: #FFF;
  border: none;
  border-radius: 26px;
  height: 46px;
  line-height: 46px;
  font-size: 1.6rem;
  padding: 0 32px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c8 span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 span svg {
  font-size: 1.2rem;
  margin-right: 8px;
}

.c9 {
  background-color: transparent;
  color: #fa5b0f;
  border: 1px solid #fa5b0f;
  border-radius: 26px;
  height: 46px;
  line-height: 46px;
  font-size: 1.6rem;
  padding: 0 32px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c9 span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 span svg {
  font-size: 1.2rem;
  margin-right: 8px;
}

<main
  class="c0"
>
  <img
    alt="Um desenvolvedor de frente para um tela com código."
    class="c1"
    src="/images/perfil_b.png"
  />
  <div
    class="c2"
  >
    <h3
      class="c3"
    >
      Fala galera!
    </h3>
    <div
      class="c4"
    >
      <h1
        class="c5"
      >
        SOU
      </h1>
      <span
        class="c6"
        id="test-span"
      >
        <div
          class="Typewriter"
          data-testid="typewriter-wrapper"
        >
          <span
            class="Typewriter__wrapper"
          />
          <span
            class="Typewriter__cursor"
          >
            |
          </span>
        </div>
      </span>
    </div>
    <p
      class="c7"
    >
      <span>
        TypeScript, ReactJS, NextJS, e Styled Components
      </span>
    </p>
    <a
      class="c8"
      color="#FFF"
    >
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"
          />
        </svg>
        MAIS SOBRE MIM
      </span>
    </a>
    <a
      class="c9"
      color="#fa5b0f"
    >
      <span>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M320 336c0 8.84-7.16 16-16 16h-96c-8.84 0-16-7.16-16-16v-48H0v144c0 25.6 22.4 48 48 48h416c25.6 0 48-22.4 48-48V288H320v48zm144-208h-80V80c0-25.6-22.4-48-48-48H176c-25.6 0-48 22.4-48 48v48H48c-25.6 0-48 22.4-48 48v80h512v-80c0-25.6-22.4-48-48-48zm-144 0H192V96h128v32z"
          />
        </svg>
        PORTFÓLIO
      </span>
    </a>
  </div>
</main>
`
